forte/data/multi_pack.py

```diff
@@ -168,7 +174,15 @@ def _validate(self, entry: EntryType) -> bool:
 
     # TODO: get_subentry maybe useless
     def get_subentry(self, pack_idx: int, entry_id: int):
-        return self.get_pack_at(pack_idx).get_entry(entry_id)
+        # fix bug/enhancement #559: use pack_idx as pack_id in new version
+        pack_array_index: int = pack_idx  # the old way
+        if Version(self.pack_version) >= Version(BACKWARD_COMPATIBLE_VER):
```

Comment:
The name `BACKWARD_COMPATIBLE_VER` is a bit confusing since we don't know what is backward compatible, and I think it should not be in the `version.py`.

Here, we should have a const in this file, probably named as `PACK_ID_VERSION`, and leave a comment right above it explaining what this version number means. And then we do the `if` check.

Reply:
As mentioned above this name is now changed to "PACK_ID_COMPATIBLE_VERSION" , (the same change (name) will also be in 500/547 related pull requests which resolves version specific issue). Related comments explaining the meaning of this version is now added.